pandas/tests/test_categorical.py

```diff
@@ -4601,6 +4601,45 @@ def f():
         df = pd.DataFrame({'a': pd.Categorical(idx)})
         tm.assert_frame_equal(df.fillna(value=pd.NaT), df)
 
+    @pytest.mark.parametrize('fill_value, expected_output', [
+        ('a', ['a', 'a', 'b', 'a', 'a']),
+        ({1: 'a', 3: 'b', 4: 'b'}, ['a', 'a', 'b', 'b', 'b']),
+        ({1: 'a'}, ['a', 'a', 'b', np.nan, np.nan]),
+        ({1: 'a', 3: 'b'}, ['a', 'a', 'b', 'b', np.nan]),
+        (pd.Series('a'), ['a', np.nan, 'b', np.nan, np.nan]),
+        (pd.Series('a', index=[1]), ['a', 'a', 'b', np.nan, np.nan]),
+        (pd.Series({1: 'a', 3: 'b'}), ['a', 'a', 'b', 'b', np.nan]),
+        (pd.Series(['a', 'b'], index=[3, 4]), ['a', np.nan, 'b', 'a', 'b'])
+    ])
+    def test_fillna_series_categorical(self, fill_value, expected_output):
+        # GH 17033
+        # Test fillna for a Categorical series
+        data = ['a', np.nan, 'b', np.nan, np.nan]
+        s = pd.Series(pd.Categorical(data, categories=['a', 'b']))
+        exp = pd.Series(pd.Categorical(expected_output, categories=['a', 'b']))
+        tm.assert_series_equal(s.fillna(fill_value), exp)
+
+    def test_fillna_series_categorical_errormsg(self):
+        data = ['a', np.nan, 'b', np.nan, np.nan]
+        s = pd.Series(pd.Categorical(data, categories=['a', 'b']))
+
```

Comment:
these tests can all go in ``pandas/tests/series/test_missing.py``. also pls move any related fillna tests that are directly on Series/DataFrame as well (to tests/dataframe/test_missing.py), if they are *only* on Categorical itself then leave here.

Reply:
Moved the new tests relating to Series to ``pandas/tests/series/test_missing.py`` and moved the existing ``test_fillna``  from ``pandas/tests/test_categorical.py`` to ``pandas/tests/frame/test_missing.py``